allen-cell-animated/timelapse-colorizer
# feature: Add annotations tab and click interaction (annotations pt. 3)

Problem
=======
Part 3 of 5(ish) for #484, prototyping annotations! This change adds in the annotations tab for editing and viewing annotations. 

This change adds in the annotations tab, and allows users to enter into an annotation edit mode. Clicking on objects will toggle them as being labeled or unlabeled. You can add and edit labels from the tab menu.

There are a LOT of quality-of-life changes that have been omitted for the sake of line count. Changes to the cursor and additional visualization of what points are selected vs. not will be made in the next PRs, parts 4 and 5. (There's a preview of where all the PRs are headed here: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-498)

Solution
========
- Adds the `AnnotationTab` component, which allows users to add and remove labels and view their currently selected IDs.
- Users can toggle on/off annotation editing mode from a button in the tab.
- When in annotation editing mode, clicking on a cell will toggle whether it is labeled or unlabeled.  
- Updates styling for a success button style.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-501/
2. Open the annotations tab and click to enable the annotation editing mode.
3. Click on cells in the viewport. You should see their ID + track number appear in the table on the right.
4. Click on the table rows to select those objects.

Screenshots (optional):
-----------------------
**Video talk-through (🔊):**

https://github.com/user-attachments/assets/c39b49d1-3804-497a-bd1e-2e0de831d26f

![image](https://github.com/user-attachments/assets/f5a24b31-dabc-48f8-8f78-1f692637c66a)

![image](https://github.com/user-attachments/assets/a7be67dc-ad10-44c2-8337-00a94281b027)


Keyfiles (delete if not relevant):
-----------------------
1. AnnotationTab.tsx
